crates/python/qcs_sdk.pyi

```diff
@@ -0,0 +1,124 @@
+from typing import Any, Awaitable, Dict, List, Optional, TypedDict
+from numbers import Number
+
+RecalculationTable = List[str]
+Memory = Dict[str, List[float]]
+PatchValues = Dict[str, List[float]]
+
+class RewriteArithmeticResults(TypedDict):
+    program: str
+    recalculation_table: RecalculationTable
+
+class TranslationResult(TypedDict):
+    memory_descriptors: Optional[Dict[str, Any]]
+    program: str
+    ro_sources: Optional[List[List[str]]]
+    settings_timestamp: Optional[str]
+
+class ExecutionResult(TypedDict):
+    shape: List[int]
+    data: List[Number]
+    dtype: str
+
+class ExecutionResults(TypedDict):
+    buffers: Dict[str, ExecutionResult]
+    execution_duration_microseconds: Optional[int]
+
+def compile(quil: str, target_device: str) -> Awaitable[str]:
+    """
+    Uses quilc to convert a quil program to native Quil.
+
+    Args:
+        quil: A quil program.
+        target_device: The device to target for nativization (e.g. "Aspen-M-2")
+
+    Returns:
+        An Awaitable that resolves to the native Quil program.
+    """
```

Comment:
Is there a particular docstring format we want to use? I went with something reminiscent of the Google standards. In `pyquil` I believe there are some Sphinx formatted docstrings, but not sure if we want to do the same here.